CRM/Member/BAO/Membership.php

```diff
@@ -1841,6 +1843,13 @@ public static function processMembership($contactID, $membershipTypeID, $is_test
     $membershipTypeDetails = CRM_Member_BAO_MembershipType::getMembershipTypeDetails($membershipTypeID);
     $dates = array();
     $ids = array();
+
```

Comment:
@mattwire after working through this I'm pretty comfortable this is done later on in the same function - https://github.com/civicrm/civicrm-core/blob/aaa7b0e65e84d3254a485958b14f65bac5bcea75/CRM/Member/BAO/Membership.php#L2019

Reply:
@eileenmcnaughton Ok I've removed the additional check